src/s_path.c

```diff
@@ -237,10 +237,12 @@ void sys_setextrapath(const char *p)
 {
     char pathbuf[MAXPDSTRING];
     namelist_free(STUFF->st_staticpath);
+    /* add built-in "extra" path first so its checked first */
+    STUFF->st_staticpath = namelist_append(0, p, 0);
```

Comment:
Wouldn't this line suffice? I think you are appending multiple copies with each append...